eng/common/pipelines/templates/steps/docker-pull-image.yml

```diff
@@ -1,15 +1,22 @@
 parameters: 
-  - name: ContainerRegistryClientId
```

Comment:
How do you plan to make this change in common without breaking the other consumers?

Reply:
This is only used in `docindex.yml` for Python and Java and used indirectly though `update-docsms-metadata.yml` (which only has `DocValidationImageId` set in java and python repos). 

https://github.com/search?q=org%3AAzure+update-docsms-metadata.yml+path%3A**%2F*.yml&type=code

This should be a straightforward change where merges happen at roughly the same time: 
* eng/common PR 
* Java PR 
* Python PR 

The rest will not include this template as `docker-pull-image.yml` is only used when `DocValidationImageId` is not empty https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/pipelines/templates/steps/update-docsms-metadata.yml#L87